cmd/aro/operator.go

```diff
@@ -226,6 +233,17 @@ func operator(ctx context.Context, log *logrus.Entry) error {
 			client)).SetupWithManager(mgr); err != nil {
 			return fmt.Errorf("unable to create controller %s: %v", cloudproviderconfig.ControllerName, err)
 		}
+
+		// only register CPMS controller on clusters that support the CRD
+		if err := discovery.ServerSupportsVersion(discoverycli, machinev1.GroupVersion); err == nil {
+			if err = (cpms.NewReconciler(
+				log.WithField("controller", cpms.ControllerName),
+				client)).SetupWithManager(mgr); err != nil {
+				return fmt.Errorf("unable to create controller %s: %v", cpms.ControllerName, err)
```

Comment:
Not really a big deal, but ultimately we want `err` as a string, may as well do it directly IMO.  

```suggestion
				return fmt.Errorf("unable to create controller %s: %s", cpms.ControllerName, err.String())
```

Reply:
This is an interesting suggestion, but it's not in line with the conventions we have been following throughout the codebase (e.g. https://github.com/Azure/ARO-RP/blob/e711e610a0ad352ac11d0e72328170d759638242/pkg/operator/controllers/workaround/workaround_controller.go#L72). I think if we want to do this, it should be done in a dedicated "style-fixup" PR.

Reply:
a lot of our code uses `%v`, but it is rare that `%v` or `%s` are best here, we should use `%w` unless we care for some reason that the Error type we return is distinct from `err` itself: https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w

`%s` vs `%v` is also discussed in that thread - the biggest concern is if `err` is nil, which shouldn't happen...?

Reply:
Thanks for the link. I forgot there was an error formatter, that would be ideal.

Reply:
The entirety of cmd/aro/operator.go is a prime candidate for some `%w` replacement on error wrapping - I think this should be done in a follow-up PR across _all_ of our controller instantiation.